### ISSUE TITLE
CNN can`t fix the random seed on cuda

### ISSUE THREAD

**Issue by bamtercelboo** [pytorch/pytorch#4340]
`nn.Conv2d` can not  fix the random seed on the GPU, but it is ok on the CPU, I try all the solutions that I can thinlk,  still not working.

**my demo:**
`self.convs1 = [nn.Conv2d(in_channels=Ci, out_channels=Co, kernel_size=(K, D), bias=True) for K in Ks]`
`ks = [3, 4, 5]`

so , others has the similar problem ? what`s the reason and  how to solve it ?

**Comment by ezyang** [comment]
How are you setting the random seed?

**Comment by bamtercelboo** [comment]
@ezyang Sure
```
torch.manual_seed(0)
random.seed(0)
```
even try:
```
torch.cuda.seed()
torch.cuda.manual_seed()
```
only `nn.Conv2d` is not working, `LSTM、GRU` is ok.

**Comment by apaszke** [comment]
Can you elaborate what's the exact problem? `nn.Conv2d` doesn't use randomness in any way, so it shouldn't interact with seeding

**Comment by jonmorton** [comment]
@bamtercelboo you did not pass any argument to `torch.cuda.manual_seed`. Try:

```
import random

random.seed(0)
torch.manual_seed(0)
torch.cuda.manual_seed(0)
```

**Comment by apaszke** [comment]
This issue has been inactive so I'll close it, but feel free to reopen if you still have the problem.

**Comment by bamtercelboo** [comment]
@apaszke Hi, I have find the bug for the problem,  the bug appear when use nn.Embedding() , max_norm is not None.
if you use like this, `self.embed = nn.Embedding(V, D, max_norm=None, scale_grad_by_freq=True)` , this is ok, no problem, gpu and cpu both ok.
but when you use it like this, `self.embed = nn.Embedding(V, D, max_norm=5, scale_grad_by_freq=True)` , the random seed can not fix in gpu, cpu is ok.
so, I wonder to know how to fix the random seed when use max_norm is not None, whether there is some error in nn.Embedding().
